ZipBuilder/Sources/ZipBuilder/ZipBuilder.swift

```diff
@@ -0,0 +1,759 @@
+    let readmePath = paths.templateDir.appendingPathComponent(Constants.ProjectPath.readmeName)
+    let readmeTemplate: String
+    do {
+      readmeTemplate = try String(contentsOf: readmePath)
+    } catch {
+      fatalError("Could not get contents of the README template: \(error)")
+    }
+
+    // Break the `subspecsToInstall` into a variable since it's helpful when debugging non-cache
+    // builds to just install a subset: `[.core, .analytics, .storage, .firestore]` for example.
+    let subspecsToInstall = Subspec.allCases()
+
+    // We need to install all the subpsecs in order to get every single framework that we'll need
+    // for the zip file. We can't install each one individually since some pods depend on different
+    // subspecs from the same pod (ex: GoogleUtilities, GoogleToolboxForMac, etc). All of the code
+    // wouldn't be included so we need to install all of the subspecs to catch the superset of all
+    // required frameworks, then use that as the source of frameworks to pull from when including
+    // the folders in each product directory.
+    CocoaPodUtils.installSubspecs(subspecsToInstall,
+                                  inDir: projectDir,
+                                  customSpecRepos: customSpecRepos)
+
+    // If any expected versions were passed in, we should verify that those were actually installed
+    // and get the list of actual versions we'll be using to build the Zip file. This method will
+    // throw a fatalError if any versions are mismatched.
+    validateExpectedVersions(inProjectDir: projectDir)
+
+    let installedPods = CocoaPodUtils.installedPodsInfo(inProjectDir: projectDir)
+    let filesToInstall = generateFrameworksWithResources(fromPods: installedPods,
+                                                         inProjectDir: projectDir,
+                                                         useCache: useCache)
+
+    // Create an array that has the Pod name as the key and the array of frameworks needed - this
+    // will be used as the source of truth for all frameworks to be copied in each product's
+    // directory.
+    let frameworks = filesToInstall.mapValues { $0.frameworks }
+    for (framework, paths) in frameworks {
+      print("Frameworks for pod: \(framework) were compiled at \(paths)")
+    }
+
+    // TODO: Overwrite the `CoreDiagnostics.framework` in the generated framework.
+
+    // Time to assemble the folder structure of the Zip file. In order to get the frameworks
+    // required, we will `pod install` only those subspecs and then fetch the information for all
+    // the frameworks that were installed, copying the frameworks from our list of compiled
+    // frameworks. The whole process is:
+    // 1. Copy any required files (headers, modulemap, etc) over beforehand to fail fast if anything
+    //    is misconfigured.
+    // 2. Get the frameworks required for Analytics, copy them to the Analytics folder.
```

Comment:
This will need to change in Firebase 6 when Analytics is not always required.

Reply:
SGTM, adding a `TODO`.